management/src/main/java/io/micronaut/management/health/indicator/service/ServiceReadyHealthIndicatorConfiguration.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.management.health.indicator.service;
+
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.core.util.Toggleable;
+
+import static io.micronaut.management.health.indicator.service.ServiceReadyHealthIndicatorConfiguration.PREFIX;
+
+/**
+ * @author Sergio del Amo
+ * @since 3.8.6
+ */
+@ConfigurationProperties(PREFIX)
+@Requires(property = PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+public class ServiceReadyHealthIndicatorConfiguration implements Toggleable {
```

Comment:
a whole new configuration class for one property seems overkill to me. This can be added to the existing `HealthEndpoint` class as a property which is already configurable https://docs.micronaut.io/latest/guide/configurationreference.html#io.micronaut.management.endpoint.health.HealthEndpoint

Reply:
Please note that we follows the same approach (a configuration class) for discovery. 

https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/management/src/main/java/io/micronaut/management/health/indicator/discovery/DiscoveryClientHealthIndicatorConfiguration.java

Reply:
which also seems overkill

Reply:
I could delete `ServiceReadyHealthIndicatorConfiguration` and annotate the indicator with an `@Requires`:

```java
@Requires(property = "endpoints.health.service.enabled", notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
public class ServiceReadyHealthIndicator implements HealthIndicator {
```

However, this will mean no configuration reference documentation is generated. Users will have no idea how to disable the indicator unless they go to the code.

Reply:
agreed, which is why it makes more sense these config options are in `HealthEndpoint` where they will show up in the docs

Reply:
As requested, I have moved the class as a static class to `HealthEndpoint`.   

https://github.com/micronaut-projects/micronaut-core/pull/8836/commits/2f7fc2250b1060138eb1fdfe5143cf3575654959

The amount of code is almost identical. It is just an inner class instead of a separate class. 

Moreover, having a separate configuration class in the same package of the indicator is a nicer design. `HealthPoint` is an aggregator of indicators without knowing anything about indicators or how the indicators are configured, or if they can be toggled.